src/types/Account/_sortPlugins.spec.js

```diff
@@ -0,0 +1,310 @@
+        [IdentitySyncWorker, true, true],
+        [dummyWorker, false, false],
+      ])
+    });
+
+    it('should handle withSinglePluginAndSingleInjectBeforeDependenciesWorker', function () {
+      const sortedPlugins = sortPlugins(accountOnlineWithDefaultPlugins, userDefinedWithSinglePluginAndSingleInjectBeforeDependenciesWorker);
+      expect(sortedPlugins).to.deep.equal([
+        [ChainPlugin, true, true],
+        [withSinglePluginAndSingleInjectBeforeDependenciesWorker, false, false],
+        [TransactionSyncStreamWorker, true, true],
+        [IdentitySyncWorker, true, true],
+        [dummyWorker, false, false],
+      ])
+    });
+
+    it('should handle userDefinedWithMultiplePluginDependenciesPlugins', async function () {
+      const sortedPlugins = sortPlugins(accountOnlineWithDefaultPlugins, userDefinedWithMultiplePluginDependenciesPlugins);
+      expect(sortedPlugins).to.deep.equal([
+        [ChainPlugin, true, true],
+        [withSinglePluginDependenciesWorker2, false, false],
+        [TransactionSyncStreamWorker, true, true],
+        [IdentitySyncWorker, true, true],
+        [withSinglePluginDependenciesWorker, false, false],
+        [dummyWorker, false, false],
+      ]);
+    });
+    it('should handle userDefinedConflictingDependencies', function () {
+      expect(() => sortPlugins(accountOnlineWithDefaultPlugins, userDefinedConflictingDependencies))
+          .to
+          .throw('Conflicting dependency order for userDefinedConflictingDependenciesWorker');
+    });
+    it('should handle userDefinedSimpleDependencyPluginDependenciesPlugins', async function () {
+      const sortedPlugins = sortPlugins(accountOnlineWithDefaultPlugins, userDefinedSimpleDependencyPluginDependenciesPlugins);
+
+      expect(sortedPlugins).to.deep.equal([
+        [ChainPlugin, true, true],
+        [pluginWithMultiplePluginDependencies, false, false],
+        [TransactionSyncStreamWorker, true, true],
+        [IdentitySyncWorker, true, true],
+        [withSinglePluginDependenciesWorker, false, false],
+        [dummyWorker, false, false],
+      ])
+    });
+
+    it('should handle userDefinedComplexPluginDependenciesPlugins', async function () {
+      // TODO: User specified wrongly sorted plugins with deps is not yet handled.
+      // rejecting with error for now.
+      expect(() => sortPlugins(accountOnlineWithDefaultPlugins, userDefinedComplexPluginDependenciesPlugins))
+          .to
+          .throw('Dependency withSinglePluginDependenciesWorker not found');
+      // const sortedPlugins = await sortPlugins(accountOnlineWithDefaultPlugins, userDefinedComplexPluginDependenciesPlugins);
+      // expect(sortedPlugins).to.deep.equal([
+      //   [ChainPlugin, true],
+      //   [TransactionSyncStreamWorker, true],
+      //   [IdentitySyncWorker, true],
+      //   [dummyWorker, true],
+      //   [withSinglePluginDependenciesWorker, true],
+      //   [pluginWithMultiplePluginDependencies, true],
+      // ])
```

Comment:
Usually, we don't commit commented code.

Reply:
It will stay, scope time increase if I deal with the expressed "TODO" section. 
But knowing the output of the test is required as this function has been done using TDD. 
I however can deal with it by completing the TODO, but it would a pretty consequent increase of scope (half a day of work I would guess)